examples/MQ/pixelDetector/src/PixelGeo.cxx

```diff
@@ -26,13 +27,19 @@ const char* PixelGeo::getModuleName(Int_t m)
       ASCII file should start with Pixel otherwise they will
       not be constructed
   */
-    sprintf(modName, "Pixel%i", m + 1);
+    int result_length = snprintf(modName, maxbuf - 1, "Pixel%i", m + 1);
+    if (!(result_length > 0 && result_length < static_cast<int>(maxbuf))) {
+        LOG(fatal) << "Buffer overrun in snprintf.";
+    }
     return modName;
 }
 
 const char* PixelGeo::getEleName(Int_t m)
 {
     /** Returns the element name of Det number m */
-    sprintf(eleName, "Pixel%i", m + 1);
+    int result_length = snprintf(eleName, maxbuf - 1, "Pixel%i", m + 1);
```

Comment:
Please use `sizeof(buffer)` so that is obvious that you're targeting the right size. And there is no need for `- 1` with `snprintf`.

```suggestion
    int result_length = snprintf(eleName, sizeof(eleName), "Pixel%i", m + 1);
```

BTW: The return value of `snprintf` is not the length of the result, but the length that the full result would need. So maybe a better name would be good?

And please consider simplifying the check by checking the error condidtions explicitly. Like `(retval < 0) || (retval >= sizeof(eleName)`?